src/transformation/visitors/switch.ts

```diff
@@ -1,61 +1,89 @@
 import * as ts from "typescript";
-import { LuaTarget } from "../../CompilerOptions";
 import * as lua from "../../LuaAST";
 import { FunctionVisitor } from "../context";
-import { unsupportedForTarget } from "../utils/diagnostics";
 import { performHoisting, popScope, pushScope, ScopeType } from "../utils/scope";
 
-export const transformSwitchStatement: FunctionVisitor<ts.SwitchStatement> = (statement, context) => {
-    if (context.luaTarget === LuaTarget.Universal || context.luaTarget === LuaTarget.Lua51) {
-        context.diagnostics.push(unsupportedForTarget(statement, "Switch statements", LuaTarget.Lua51));
+const containsBreakStatement = (statements: ts.Node[]): boolean => {
+    for (const s of statements) {
+        if (
+            ts.isSwitchStatement(s) ||
+            ts.isWhileStatement(s) ||
+            ts.isDoStatement(s) ||
+            ts.isForStatement(s) ||
+            ts.isForInStatement(s) ||
+            ts.isForOfStatement(s)
+        ) {
+            // Ignore: Break statements are valid as children of these
+            //         statements without breaking the clause
+        } else if (ts.isBreakStatement(s)) {
+            return true;
+        } else if (containsBreakStatement(s.getChildren())) {
+            return true;
+        }
     }
 
+    return false;
+};
+
+export const transformSwitchStatement: FunctionVisitor<ts.SwitchStatement> = (statement, context) => {
     const scope = pushScope(context, ScopeType.Switch);
 
     // Give the switch a unique name to prevent nested switches from acting up.
     const switchName = `____switch${scope.id}`;
     const switchVariable = lua.createIdentifier(switchName);
 
-    let statements: lua.Statement[] = [];
+    // Collect the fallthrough bodies for each case as defined by the switch.
+    const caseBody: lua.Statement[][] = [];
```

Comment:
So if I understand correctly: you copy/duplicate the fallthrough statements into individual clauses without any fallthrough? Might be nice to add that as a comment, maybe even with a small 2-clause example like:
```ts
// case A: 
//    foo(); 
// case B: 
//    bar(); 
//    break;
// ->
// if (A) { 
//     foo();
//     bar();
// } else if (B) { 
//     bar(); 
// }
```

Reply:
Correct. Although, after looking at some more uses of a switch in the wild, this method can create more code duplication than anticipated when a return statement is used over a break.